### PR TITLE
WIP: Process doc 

### LINKED ISSUE #13: v3 spec development process & milestones
Repurposing this issue to discuss process and milestones for development of the v3 protocol spec.

Straw man proposal for discussion:

* **Phase 1 (current)**
  * Continue discussing technical points in the current editor's draft via zarr-specs github and the v3 spec dev calls 
  * Accumulate some early implementation experience from zarr-python (Matthias), zarrita (Alistair), and at least one other language (ideally statically-typed)
  * Take tentative positions on any remaining open questions, update and complete the v3 protocol and associated specs, highlighting any particular points where we'd like feedback
  * **Milestone: Publish "First Working Draft"**
  * Request for comments
* **Phase 2**
  * Accumulate some more implementation experience, continue technical discussions
  * Compile and discuss comments from community review on first working draft, decide how to address comments
  * Revise v3 protocol spec
  * Establish a test suite for verifying completeness and compatibility between different implementations
  * **Milestone: Publish "Second Working Draft"**
  * Call for implementations
* **Phase 3**
  * Focus on implementations, getting completeness and compatibility 
  * Deal with any further (hopefully minor) technical issues that arise from new implementations
  * Finalise v3 protocol and associated specs
  * Request approval from Zarr Steering Group to publish 
  * **Milestone: Publish "Recommendation"**

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR contains initial work on a document describing the specification development process.

Resolves #13.

### CODE REVIEW

**Comment by alimanfoo** [comment]
I've done some rethinking on this and pushed some changes, still incomplete but would appreciate feedback.

@joshmoore doc now partially answers your question about versioning, but still more to discuss. Don't yet have answers for your other questions.

File: docs/process.rst [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+is slightly different from other specs.
+
+Versioning
+~~~~~~~~~~
+
+The Zarr core protocol spec is versioned using a [major].[minor]
+versioning scheme. The spec documents live in the zarr-specs GitHub
+repository at the location ``docs/protocol/core/v[major].[minor].rst``.
+
+A new major version of the core protocol spec can include change which
```
Comment (joshmoore) [inline docs/process.rst]:
```suggestion
A new major version of the core protocol spec can include changes which
```

**Comment by rabernat** [comment]
This proposal would be superseded by the ZEP process, described in https://github.com/zarr-developers/governance/pull/16.

We should review this and decide if there is anything useful here that can be used in the ZEP framework.

cc @zarr-developers/steering-council

**Comment by joshmoore** [comment]
Closing in favor of https://github.com/zarr-developers/governance/pull/16

@MSanKeys963 has reviewed and will reference it when adding, e.g. Codecs and Stores to SPEC ZEPs.

**Pull request closed without merging**